active/0000-struct-literals.md

```diff
@@ -0,0 +1,150 @@
+The only other use of `:` is in struct literals and pattern matching on structs.
+The other reason is that it might interfere with other future features like type ascription.
+
+# Motivation
+
+Because `:` is used for types, having general type ascription would be confusing.
+
+Consider this:
+
+```rust
+fn foo(Foo { a: Bar, .. }: Foo<Bar>) { ... }
+```
+
+Currently this would bind a variable to `Bar` because it's a pattern on a struct.
+Type ascription would then be something like:
+
+```rust
+fn foo(Foo { a: b: Bar, .. }: Foo<Bar>) { ... }
+```
+
+This doesn't look as good and raises questions about associativity of `:`
+
+Another feature that the current syntax prevents is potential for anonymous record types.
+By extension, this also prevents using that same syntax for keyword arguments.
+
+```rust
+x = {a: b};
+```
+
+Is this a block with type ascription or an anonymous record?
+
+
+# Detailed design
+
+Pick a syntax for structs. This syntax will be the One True Syntax for everything remotely struct related.
+There are two requirements:
+
+1. It is not ambiguous with anything currently used
+2. It is nice to use in patterns
+
+`let Point { x = a, y = b } = foo;` is probably unclear because it looks like `x` is being assigned to.
+For this reason I would not suggest `=` as in previous closed RFCs.
+
+If syntax like `Point { .x = 1, .y = 2 }` is picked, then it is quite natural that patterns would be something like:
+
+```rust
+Point { .x = a, .y = b} = foo;
+```
+
+it is clear that it is some `foo.x` that is being destructured to `a` because it `.x` is not a valid identifier.
+The current syntax is not quite as clear:
+
+```rust
+let Point { x: a, y: b } = foo;
+```
+
+While there is only one logical way to write this, I still write it backwards sometimes.
+This is because I am so used to writing
+
+```rust
+let a: i32 = bar;
+```
+
+so I always write `a: ` first in `let`s.
+
+Another available syntax is `=>` which is used in keyword arguments in some languages and in hashtables in others.
+
+# Drawbacks
+
+Currently the use follows declaration
+
+```rust
+struct Point {
+    x: i32,
+    y: i32,
+}
+
+let Point{x: first, y: second} = Point{ x: 1, y: 2};
+```
+
+the same way that 
+
+```rust
+struct Color(f32, f32, f32);
+
+let Color(red, green, blue) = Color(1.0, 1.0, 1.0);
+```
+
+where the types are replaced by their values
+
+the counter-examples in Rust are functions
+
+```rust
+fn foo(x: i32) -> i32{
+    x
+}
+
+let y = foo(5);
+```
+
+if functions uses followed declaration you'd expect `let x = foo(x: y);`
+
+But it turns out there's symmetry, especially easily seen with the `=>` syntax:
+
+```rust
+struct Color(f32, f32, f32);
+
+let Color(red: f32, green: f32, blue: f32) = Color(1.0, 1.0, 1.0);
+
+fn foo(x: i32) -> i32{
+    x
+}
+
+let y = foo(5: i32);
+
+struct Point {
+    x: i32,
+    y: i32,
+}
+
+let Point{ x: i32 => first, y: i32 => second} = Point{ x: 1, y: 2};
```

Comment:
The way I thought of it would look like:

```rust
struct Point {
    x => i32,
    y => i32,
}

let Point{x => first: i32, y => second: i32} = Point{x => 1, y => 2};
```

But in general, we are on the right track with this RFC.

You could throw some `assert!`s here and there to show what is the resulting state.

```rust
assert!(first == 1);
assert!(second == 2);
```